.github/workflows/build-mixxx.yml

```diff
@@ -0,0 +1,110 @@
+name: build Mixxx
+
+on: [pull_request, push]
+
+env:
+  #TODO: Set -DDEBUG_ASSERTIONS_FATAL=OFF before deploying CI builds as releases!!!
+  CMAKEFLAGS: -DCMAKE_BUILD_TYPE=Release -DBATTERY=ON -DBROADCAST=ON -DBULK=ON -DDEBUG_ASSERTIONS_FATAL=ON -DHID=ON -DLILV=ON -DOPUS=ON -DQTKEYCHAIN=ON -DVINYLCONTROL=ON
+  # GitHub Actions automatically zstd compresses caches
+  CCACHE_NOCOMPRESS: true
+
+jobs:
+  build-mac:
+    name: macOS
+    runs-on: macos-10.15
+    env:
+      MIXXX_ENVNAME: 2.3-1db6c06-sdk10.15-macosminimum10.12-x86_64
+      # macOS 10.12 is the minimum requirement for Qt 5.12.
+      # Check the minimum requirement when updating to a build environment
+      # with an updated Qt version.
+      MACOSX_DEPLOYMENT_TARGET: 10.12
+      CMAKEFLAGS_EXTRA: -DCOREAUDIO=ON -DHSS1394=ON -DMACOS_BUNDLE=ON
+      GTEST_COLOR: 1
+      CTEST_OUTPUT_ON_FAILURE: 1
+      # Render analyzer waveform tests to an offscreen buffer
+      QT_QPA_PLATFORM: offscreen
```

Comment:
This can be moved to a separate ctest step

Reply:
Unfortunately it cannot. When I tried to do that, every test SIGABORTed.

Reply:
So this does not work? https://github.com/mixxxdj/mixxx/blob/2.3/.github/workflows/build.yml#L95L101

Reply:
I'll give it another try, but when I tried before it just crashed.

Reply:
I tried again and it [failed](https://github.com/mixxxdj/mixxx/runs/1420695792?check_suite_focus=true#step:7:3748).

Reply:
Weird.